tests/test_packages/test_protocols/test_fipa.py

```diff
@@ -290,16 +288,217 @@ def test_fipa_decoding_unknown_performative():
             FIPASerializer().decode(encoded_msg)
 
 
-def test_dialogues():
-    """Test the dialogues model."""
-    dialogues = FIPADialogues()
-    result = dialogues.create_self_initiated(dialogue_starter_addr="starter", dialogue_opponent_addr="opponent", is_seller=True)
-    assert isinstance(result, FIPADialogue)
-    result = dialogues.create_opponent_initiated(dialogue_opponent_addr="opponent", dialogue_reference=(str(0), ''), is_seller=False)
-    assert isinstance(result, FIPADialogue)
-    assert result.role == FIPADialogue.AgentRole.BUYER
-    assert dialogues.dialogue_stats is not None
-    dialogues.dialogue_stats.add_dialogue_endstate(FIPADialogue.EndState.SUCCESSFUL, is_self_initiated=True)
-    dialogues.dialogue_stats.add_dialogue_endstate(FIPADialogue.EndState.DECLINED_CFP, is_self_initiated=False)
-    assert dialogues.dialogue_stats.self_initiated == {FIPADialogue.EndState.SUCCESSFUL: 1, FIPADialogue.EndState.DECLINED_PROPOSE: 0, FIPADialogue.EndState.DECLINED_ACCEPT: 0, FIPADialogue.EndState.DECLINED_CFP: 0}
-    assert dialogues.dialogue_stats.other_initiated == {FIPADialogue.EndState.SUCCESSFUL: 0, FIPADialogue.EndState.DECLINED_PROPOSE: 0, FIPADialogue.EndState.DECLINED_ACCEPT: 0, FIPADialogue.EndState.DECLINED_CFP: 1}
+class Test_dialogues:
+    """Tests dialogues model from the packages protocols fipa."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set up the test."""
+        cls.client_dialogues = FIPADialogues()
+        cls.seller_dialogues = FIPADialogues()
+
```

Comment:
I'd add two more variables here for clarity:
`cls.client_addr = "client"` and `cls.seller_addr = "seller"`, then use them throughout. We want to avoid magic strings and numbers as they invite for errors